### PR TITLE
Handle musescore midis

### PR DESCRIPTION
**Pull request by jackkinsella**

### CODE REVIEW

**Comment by jackkinsella** [comment]
How does this sound for the Brahms? 

(I know the code is getting messy and needs to be split up into a few files, but for this PR the goal is to merge as soon as we have the bug fixed)

File: midi-to-part-mp3s [hunk 1]
```diff
@@ -166,8 +172,48 @@ def generate_full_mp3(solo_parts: List[Part]) -> None:
   output_file_path = "{}/all.mp3".format(output_directory)
   combiner.build(input_files, output_file_path, 'mix-power')
 
+def has_separate_tempo_map(number_of_tracks: int) -> bool:
+    if number_of_tracks == 4:
```
[the remark this answers is not in the document]
Comment (jackkinsella) [reply midi-to-part-mp3s]: Yea you're right, that would be a better approach — +1

**Comment by jackkinsella** [comment]
Thanks for the feedback — will take another stab at this tomorrow. I feel we're slowly getting there. 

Something I noticed is that it seems to be only the soprano track that is timed incorrectly in the Brahms (i.e. the other three voices play in sync). This narrows down the search space somewhat.

**Comment by oli-ver** [comment]
Yes exactly: That was what I meant with my comment:

> Is it possible that we do not only have to extract the program change messages from the first track to another track but that we need to remove them from the other track afterwards?

The meta messages in the soprano track will now exist two times only for soprano won't they?

**Comment by jackkinsella** [comment]
Could be... will check.

File: midi-to-part-mp3s (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackkinsella) [reply midi-to-part-mp3s]: (I will rewrite this way tomorrow.)

**Comment by jackkinsella** [comment]
I tried trimming the `control_change` and `program_change` events from track1 voice in the Brahms, but the problem persisted.

**Comment by jackkinsella** [comment]
Another potential line of inquiry – I wonder if the change in time signature in the 2nd bar somehow contributes to the timing issues:
Screen Shot 2019-06-17 at 19 26 42

**Comment by jackkinsella** [comment]
FYI here is a printout of the first few midi messages in the original:
Screen Shot 2019-06-17 at 19 28 37

And, from listening to the Soprano solo, you can hear where it goes wrong in the third note:
[soprano.mp3.zip](https://github.com/jackkinsella/midi-to-part-mp3s/files/3298162/soprano.mp3.zip)

**Comment by oli-ver** [comment]
Just to let you know: I will not be able to work on this until Friday but send some comments now or then. So do you think the midi / MusicXML of the Brahms is faulty?

**Comment by jackkinsella** [comment]
Totally cool about you not being available (I don't expect anything in open source to happen even on a weekly basis). A lot of this is me debugging aloud so don't feel any pressure to comment :)

The file broken? Hard to say. But one thing I know: most midi players can play the whole accurately. E.g. It works perfectly in three different midi players I tried.

I also don't know much about how midi works so I might dive into the specification and also check out the code for a reference player to see what I learn.

There's also the possibility that `mido`, the python midi library has an issue (or that I've been making a mistake in how I'm using it).

**Comment by jackkinsella** [comment]
FIXED IT! I realized what the problem was while I was making breakfast — basically the tempo track (track 0) wasn't being used in every intermediary part in the audio production pipeline.
[all.mp3.zip](https://github.com/jackkinsella/midi-to-part-mp3s/files/3304972/all.mp3.zip)

**Comment by jackkinsella** [comment]
Also managed to get rid of the clunky old `common-solo-tracks` bit.

**Comment by oli-ver** [comment]
Nice, great progress! And also good that we don't need the --common-solo-tracks parameter anymore

**Pull request merged**